examples/protected_base/construct.yaml

```
@@ -0,0 +1,22 @@
+# yaml-language-server: $schema=../../constructor/data/construct.schema.json
+"$schema": "../../constructor/data/construct.schema.json"
+
+name: ProtectedBaseEnv
+version: X
+installer_type: all
+
+channels:
+  - defaults
+
+specs:
+  - python
+  - conda
+
+extra_envs:
+  default:
+    specs:
+      - python
+      - pip
+
+extra_files:
+  - frozen.json: conda-meta/frozen
```

```suggestion
extra_files:
  - frozen.json: conda-meta/frozen
  - frozen.json: envs/defaults/conda-meta/frozen
```

Since the extra environments are handled in separate code blocks than the `base` environment for SH and PKG, I think the `default` environment needs to be frozen, too, to get more test coverage. `test_frozen_environment` needs to be adjusted to test for that `frozen` file, too, in this case.